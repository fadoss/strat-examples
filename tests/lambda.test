semantics/lambda

smod MAIN is
	protecting MAIN-BASE .
	extending LAMBDA-REDUCE{Normal} .
endsm

srew K I Omega using fullReduce .
srew ((\ x . v(1)) (\ y . v(2))) ((\ x . v(3)) (\ y . v(4))) using fullReduce .

smod MAIN is
	protecting MAIN-BASE .
	extending LAMBDA-REDUCE{Applicative} .
endsm

srew ((\ x . v(1)) (\ y . v(2))) ((\ x . v(3)) (\ y . v(4))) using applicative .
srew A* churchNum(4) (A+ churchNum(2) churchNum(1)) using fullReduce .
