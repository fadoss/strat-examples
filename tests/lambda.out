==========================================
srewrite in MAIN : K I Omega using fullReduce .

Solution 1
rewrites: 17
result LambdaTerm: \ x . x

No more solutions.
rewrites: 17
==========================================
srewrite in MAIN : \ x . v(1) \ y . v(2) (\ x . v(3) \ y . v(4)) using
    fullReduce .

Solution 1
rewrites: 8
result LambdaTerm: v(1) v(3)

No more solutions.
rewrites: 8
==========================================
srewrite in MAIN : \ x . v(1) \ y . v(2) (\ x . v(3) \ y . v(4)) using
    applicative .

Solution 1
rewrites: 4
result LambdaTerm: \ x . v(1) \ y . v(2) v(3)

No more solutions.
rewrites: 4
==========================================
srewrite in MAIN : A* churchNum(4) (A+ churchNum(2) churchNum(1)) using
    fullReduce .

Solution 1
rewrites: 345
result LambdaTerm: \ z . \ q . \ x . (z (z (z (z (z (z (z (z (z (z (z (z
    x))))))))))))

No more solutions.
rewrites: 345
Bye.
